profile-page ui done

profe-ssor opened the pull request:
User Profile Fetch Logic:

    A new method getById was added to the user SService, which interacts with the backend to fetch user data based on their ID.

Dummy Data Added:

    Dummy data for profession, country, city, postalCode, and email has been added to the API interfaces for testing purposes.

ProfilePageComponent Update:

    The ProfilePageComponent now initializes the form with the retrieved user data (from  user Service), and the fields are pre-populated when the user navigates to the profile page.

Joseph-Ajegetina reviewed (commented):
is this the page ? 
![image](https://github.com/user-attachments/assets/ceec0bb6-08fe-4329-abc4-8de0ab24ed62)

Joseph-Ajegetina commented:
Connect the actions to the backend 
So a user should be able to 

* upload a profile picture 
* edit basic info 
* toast to confirm success

profe-ssor commented:
1. implemented the api to update fields, upload and save  images

2. Added onImageSelected method to update the image 

3. Added getProfileImageUrl method to get or accept the image url if on the loacal machine or uploaded to the server 

3. Added saveProfile nethod to save the edited fileds
